src/monsters.cpp

```diff
@@ -124,10 +124,13 @@ bool Monsters::deserializeSpell(const pugi::xml_node& node, spellBlock_t& sb, co
 		          << " - Missing chance value on non-melee spell: " << name << std::endl;
 	}
 
+	sb.range = std::max<uint32_t>(Map::maxClientViewportX + 1, Map::maxClientViewportY + 1);
 	if ((attr = node.attribute("range"))) {
 		uint32_t range = pugi::cast<uint32_t>(attr.value());
-		if (range > (Map::maxViewportX * 2)) {
-			range = Map::maxViewportX * 2;
+		if (range > std::max<uint32_t>(Map::maxClientViewportX + 1, Map::maxClientViewportY + 1)) {
+			range = std::max<uint32_t>(Map::maxClientViewportX + 1, Map::maxClientViewportY + 1);
+			std::cout << "[Warning - Monsters::deserializeSpell] - " << description
```

Comment:
IMO it should either:
1. Warn the user of the mistake, and allow it to proceed - if it does not crash the server
2. Forbid using invalid range values, and refuse to proceed (return false)
But not change the value to something the user explicitly set otherwise.

Reply:
Changing the value has no impact because its constrained to the creatures max view. I do it to keep the number clean and sanitized for wherever its used in the future. Also this is done in other parts of the code both warning + setting the value (IE when the chance is parsed). If we know how to fix the value on the users behalf we should do that and log a warning.